packages/core/src/utils/react-query-utils.ts

```diff
@@ -20,6 +20,21 @@ export interface QueryCacheFunctions<T> {
 
 export const getQueryCacheFunctions = <T>(queryKey: QueryKey): QueryCacheFunctions<T> => ({
   mutate: (newData, opts = {refetch: true}) => {
+    // Shim from https://developers.google.com/web/updates/2015/08/using-requestidlecallback
+    const requestIdleCallback =
+      window.requestIdleCallback ||
+      function (cb: any) {
+        var start = Date.now()
+        return setTimeout(function () {
+          cb({
+            didTimeout: false,
+            timeRemaining: function () {
+              return Math.max(0, 50 - (Date.now() - start))
+            },
+          })
+        }, 1)
+      }
+
```

Comment:
Awesome, thank you @jamiedavenport! Just one thing, can you move this shim out of `getQueryCacheFunctions` up to the top level?

Reply:
Updated